### ISSUE TITLE
Add missing `contributors` field to import schema

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#8717]
We have a schema defining the records our import endpoint supports here: https://github.com/internetarchive/openlibrary-client/blob/master/olclient/schemata/import.schema.json

This schema should also support fields like:

```json
{
    "contributors": [
        { "role": "Editor", "name": "Foo bar" }
    ]
}
```

Update the schema document to support these. See how the `author` field works in that file as an example.

### Additional context
https://github.com/internetarchive/openlibrary-client/blob/master/olclient/schemata/import.schema.json

### Stakeholders
@Billa05

**Comment by hornc** [comment]
How does this import field relate to the existing `contributions` string field on Editions: https://github.com/internetarchive/openlibrary-client/blob/7f537ff7f904d2be26a664c004cc49c18aad94db/olclient/schemata/edition.schema.json#L140
and https://openlibrary.org/type/edition ?

or the `author_role` on Works? https://openlibrary.org/type/author_role

It looks like some code may expect or add `contributors` (I'm not 100% sure), but it's not documented in the same way as `contributions`, and there seems to be some overlap between `contributors`, `contributions`, and `author_role`.

**Comment by cdrini** [comment]
They are annoyingly conflated, unfortunately. All three are used extensively in our db. Here are the differences as I see them:

1. `contributions` - Just a list of strings
2. `contributors` - List of `{ role: str, name: str }` objects (I prefer this one to `contributions` since it lets us include `role`. But note role is just a plain string!
3. `authors` - List of `{ type: ThingReference[ThingType], author: ThingReference[Author] }` 
    * e.g. `{"type": {"key": "/type/author_role"}, "author": {"key": "/authors/OL33686A"}}`
    * Note: the only `type` that appears in our database is `/type/author_role`
    * I believe the original intent was to add new types for every kind of "role"? E.g. `editor_role`, etc. But that never happened.

Long term, I believe we should move to something that's at the intersection of `contributors` and `authors`.

Short term though, the Import schema should just be aware that the `contributors` syntax is also supported.

**Comment by hornc** [comment]
@cdrini I don't think new types were intended along with `author_role`, see https://openlibrary.org/type/author_role

That looks to me like you could have an `author_role` of `author` =  `OL123A` and `role` = `editor` (a string).

The `as` looks like it could handle aliases and pen-names etc.  I think the `author_role` works for _any_ contributor, because OL models any individual or organisation that is responsible for creating a book as an `Author`, so every attribution a OL....A is an `Author` by definition, but `author_role` lets us be more precise if they are an editor, illustrator, or whatever else.   

I agree that this isn't being used like this now, and also that a better solution would be to move towards using `author_role`.

Also, I wasn't aware that `contributors` syntax was valid, I was going to argue that it wasn't because it's not documented, but I saw that it is already used in the code in numerous places, so I'm too late :) 

It looks like the MARC import path will populate `contributions` based on strings found within MARC records in 7xx fields (the exact role is not always present as a string), and non-MARC imports might use the name + role `contributors`.

The code for the MARC `contributons` is here: https://github.com/internetarchive/openlibrary/blob/52725163c497f1bb71fedae511aa2737b49543c6/openlibrary/catalog/marc/parse.py#L547

The 7xx fields are a bit vague, they are for people or orgs "having some form of responsibility for the creation of the work"
https://www.loc.gov/marc/bibliographic/bd70x75x.html .   In many cases they almost look like co-authors, or authors of part of a work, but it also seem like many other relationships could be shown, and in practice it's not necessarily specified.

One of the examples includes `700 	1#$aJefferson, Thomas,$d1743-1826,$eformer owner.$5MH ` so that's not really a contributor (unless he's being credited as the "author" of the book plate or ownership inscription.... or marginalia)

After looking into it, I'm not very convinced the string `contributions` from MARC is adding much. It's likely to be a bit misleading in many cases, and when it's accurate, it's still not clear exactly what is means. It's just an associated name string pulled from the MARC without the possibility of extra context that may or may not be present in the source. :( I don't currently have a better suggestion on what to do with `7xx` fields from MARC records though. I think the current justification is: at least we are copying a name string across if it exists and adding it somewhere (to aid identification).

The other problems we'll have is:
* `author_role`s belong on Works, but `contributors` belong on Editions
* `author_role` requires an OLID, OL....A, so we need to do matching and author creation if we don't find one...


I have found a handful of works that use the `role` parameter, I think that's _all_ that exist:

/works/OL15079095W
/works/OL20495341W
/works/OL9323420W
/works/OL235841W
/works/OL9323410W
/works/OL9675857W
/works/OL12631165W

In conclusion:
* I now agree `contributors` ({ role: str, name: str })  seems better than just the str one.
* Deprecating the MARC sourced `contributions` to use `contributors` with a role will take a bit of thought, basically it means using MARC 7xx fields effectively and reviewing that whole area of authors / contributors.
* I think we need some clarity on what contributors belong on a Work and which ones on an Edition. (examples!)

**Comment by cdrini** [comment]
Ah you are correct! `type` is the type of the object itself. So to update the model:

1. `contributions` - Just a list of strings
    * Added programmatically it appears. Not displayed anywhere in the UI
2. `contributors` - List of `{ role: str, name: str }` objects 
    * I prefer this one to `contributions` since it lets us include `role`. But note role is just a plain string!
    * This is what is used when you add contributors via UI
    * Displayed in the UI
    * Example: Translator on https://openlibrary.org/books/OL26752422M/La_Premi%C3%A8re_Le%C3%A7on_du_Sorcier
4. `authors` - List of `{ type: { key: "/type/author_role" }, author: ThingReference[Author], as: str, role: str }` 
    * e.g. `{"type": {"key": "/type/author_role"}, "author": {"key": "/authors/OL33686A"}}`
    * No way to add `as` or `role` via the UI. This info is also not displayed anywhere.

Overall it seems like this issue has reached a conclusion -- so you are free to open a PR @Billa05 !

**Comment by Billa05** [comment]
Hey, @cdrini, I have added the field. Could you please review it
